modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
     }
   }
 
+  /**
+   * Decrements all injuries of this Actor
+   *
+   * @return {Promise<void>}
+   */
   async decrementInjuries() {
-    this.injuries.forEach(i => this.decrementInjury(i))
+    for (let i of this.injuries)
+      await this.decrementInjury(i);
   }
 
+  /**
+   * Decrements a `duration` value for the Injury.
+   * If `duration` reaches 0, inform about Injury being finished.
+   *
+   * @param {ItemWfrp4e}injury
+   *
+   * @return {Promise<*>}
+   */
   async decrementInjury(injury) {
     if (isNaN(injury.system.duration.value))
       return ui.notifications.notify(game.i18n.format("CHAT.InjuryError", { injury: injury.name }))
 
-    injury = duplicate(injury)
-    injury.system.duration.value--
+    injury = foundry.utils.duplicate(injury);
+    injury.system.duration.value--;
 
-    if (injury.system.duration.value < 0)
+    if (injury.system.duration.value <= 0) {
       injury.system.duration.value = 0;
-
-    if (injury.system.duration.value == 0) {
-      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll")
-      chatData.speaker = { alias: this.name }
-      ChatMessage.create(chatData)
+      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll");
+      chatData.speaker = { alias: this.name };
+      ChatMessage.create(chatData);
     }
-    this.updateEmbeddedDocuments("Item", [injury]);
-  }
 
+    await this.updateEmbeddedDocuments("Item", [injury]);
+  }
 
+  /**
+   * Decrements all diseases of this Actor
+   *
+   * @return {Promise<void>}
+   */
   async decrementDiseases() {
-    this.diseases.forEach(d => this.decrementDisease(d))
+    for (let d of this.diseases)
```

Comment:
Changing `forEach` to `for…of` allows async function to properly `await` all execution

Reply:
Might be worth changing `decrementDisease` to return the modified object data, then calling `updateEmbeddedDocuments` to get it all done in a single database call, but that might need to wait until diseases utilize the new effect system.

Reply:
That might be worth looking into, however for now I just wanted to fix the obvious errors I was getting and prepare code to be more easily maintained in the future. I did it for this branch because I know that this will most likely need more scrutiny when the new effects are done.